library/src/main/java/io/ona/kujaku/views/KujakuMapView.java

```diff
@@ -585,9 +585,7 @@ private void afterStyleLoadedOperations(@NonNull Style style) {
 
         addWmtsLayers();
 
-        if (PermissionsManager.areLocationPermissionsGranted(getContext())) {
-            mapboxLocationComponentWrapper.init(KujakuMapView.this.mapboxMap, getContext());
-        }
+        mapboxLocationComponentWrapper.init(KujakuMapView.this.mapboxMap, getContext());
```

Comment:
Why remove this check?

It should only be initialized if permissions are set.

https://docs.mapbox.com/android/maps/overview/location-component/#activating

Reply:
OK, does this code need the permissions? The assumption is that this code is not supposed to use it's own location engine. Otherwise that means we have two location engines operating at the same time

Reply:
I'm not certain why location permissions are requested and therefore chose to leave it that way.

Are you certain it's not needed? Any references?

Reply:
Yea, it's not needed:

1. There are no operations that require the location permissions for the location component if you are not using the default location engine. In this case a `null` location engine
2. The current code on master uses the default location engine. If you remove the location check before initialising, you will see an exception in your logs because it tries to access the location. Once you disable the default location engine, it no longer tries to perform any location related operations
3. This is what happens in the code once you disable the location engine https://github.com/mapbox/mapbox-gl-native/blob/master/platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/location/LocationComponent.java#L242 and 
https://github.com/mapbox/mapbox-gl-native/blob/8cc100bfd56955c629d5da0bc49ce2bdbae52946/platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/location/LocationComponent.java#L353 and 
https://github.com/mapbox/mapbox-gl-native/blob/8cc100bfd56955c629d5da0bc49ce2bdbae52946/platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/location/LocationComponent.java#L424 and 
https://github.com/mapbox/mapbox-gl-native/blob/8cc100bfd56955c629d5da0bc49ce2bdbae52946/platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/location/LocationComponent.java#L242 and finally the interaction with the location engine https://github.com/mapbox/mapbox-gl-native/blob/8cc100bfd56955c629d5da0bc49ce2bdbae52946/platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/location/LocationComponent.java#L872

Reply:
OK, thanks.